### PR TITLE
v1.17 cherry-pick: CVE fixes

### PR DESCRIPTION
**Pull request by karlkfi**
Bug fixes in separate PR: https://github.com/GoogleContainerTools/kpt-config-sync/pull/1095

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,9 +37,9 @@ OUTPUT_DIR := $(abspath .output)
 GO_DIR := $(OUTPUT_DIR)/go
 
 # Base image used for all golang containers
-GOLANG_IMAGE := golang:1.21.3-bullseye
+GOLANG_IMAGE := golang:1.21.5-bullseye
 # Base image used for debian containers
-DEBIAN_BASE_IMAGE := gcr.io/gke-release/debian-base:bookworm-v1.0.0-gke.1
+DEBIAN_BASE_IMAGE := gcr.io/gke-release/debian-base:bookworm-v1.0.0-gke.3
```
Comment (victorpras) [inline Makefile:42]: recently I updated this image in git-sync from `gke.3` to `gke.4` in gkecl/902168 to address CVEs. I wonder why this same image is not reporting any CVE?

Comment (karlkfi) [reply Makefile:42]: Wanna make a PR on main to update to gke.4 here just to be consistent? Once it merges I can add it to this PR.

Comment (victorpras) [reply Makefile:42]: https://github.com/GoogleContainerTools/kpt-config-sync/pull/1098

Comment (karlkfi) [reply Makefile:42]: > I wonder why this same image is not reporting any CVE?

I don't think we test for CVEs in PR checks. It's only done periodically.

Comment (karlkfi) [reply Makefile]: Added #1098 to this PR

**Review by sdowell — APPROVED** [review]
/lgtm

**Pull request merged**